packages/x-charts/src/ChartsAxisHighlight/ChartsAxisHighlight.tsx

```diff
@@ -80,11 +80,34 @@ function ChartsAxisHighlight(props: ChartsAxisHighlightProps) {
 
   const getXPosition = getValueToPositionMapper(xScale);
   const getYPosition = getValueToPositionMapper(yScale);
+
+  const axisX = axis.x;
+  const axisY = axis.y;
+
+  const isBandScaleX = xAxisHighlight === 'band' && axisX !== null && isBandScale(xScale);
+  const isBandScaleY = yAxisHighlight === 'band' && axisY !== null && isBandScale(yScale);
+
+  if (process.env.NODE_ENV !== 'production') {
+    const isXError = isBandScaleX && xScale(axisX.value) === undefined;
+    const isYError = isBandScaleY && yScale(axisY.value) === undefined;
+
+    if (isXError || isYError) {
+      console.error(
+        [
+          `MUI X Charts: The position value provided for the axis is not valid for the current scale.`,
+          `This probably means something is wrong with the data passed to the chart.`,
+          `The ChartsAxisHighlight component will not be displayed.`,
+        ].join('\n'),
+      );
+    }
+  }
+
   return (
     <React.Fragment>
-      {xAxisHighlight === 'band' && axis.x !== null && isBandScale(xScale) && (
+      {isBandScaleX && (
         <ChartsAxisHighlightPath
-          d={`M ${xScale(axis.x.value)! - (xScale.step() - xScale.bandwidth()) / 2} ${
+          // @ts-expect-error, xScale value is checked in the if statement above
+          d={`M ${xScale(axisX.value) - (xScale.step() - xScale.bandwidth()) / 2} ${
```

Comment:
```suggestion
      {isBandScaleX && xScale(axisX.value) !== undefined && (
        <ChartsAxisHighlightPath
          d={`M ${xScale(axisX.value) - (xScale.step() - xScale.bandwidth()) / 2} ${
```
or
```suggestion
      {isBandScaleX && (
        <ChartsAxisHighlightPath
          d={`M ${xScale(axisX.value)! - (xScale.step() - xScale.bandwidth()) / 2} ${
```

The current comment says "The ChartsAxisHighlight component will not be displayed." but technically it renders a path with `d="M NaN ..."`

Reply:
🙃 nice catch. Went with checking if `&& xScale(axisX.value) !== undefined`